packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -211,6 +211,17 @@ Future<Null> buildGradleProjectV1(String gradle) async {
   printStatus('Built $gradleAppOutV1 (${getSizeAsMB(apkFile.lengthSync())}).');
 }
 
+File findApkFile(String buildDirectory, String buildModeName) {
+  final String apkFilename = 'app-$buildModeName.apk';
+  File apkFile = fs.file('$buildDirectory/$apkFilename');
+  if (apkFile.existsSync())
+    return apkFile;
```

Comment:
I don't think this will handle upgrades properly. When the user upgrades to the new Gradle plugin, the old apk might still be there. Since this code checks that location first, it will keep using the old apk (from before the upgrade) forever.

The workaround is to do a clean build (clearing the build directory), but this could be frustrating for the user to diagnose; it will just look like it doesn't work.

So, better to detect which Gradle plugin we're actually using and only look at its output location. Or maybe clear both locations before doing the build?